server/src/main/java/org/opensearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -225,51 +219,18 @@ public Item(@Nullable String index, XContentBuilder doc) {
             this.xContentType = doc.contentType();
         }
 
-        /**
-         * Constructor for a given item / document request
-         *
-         * @param index the index where the document is located
-         * @param type the type of the document
-         * @param id and its id
-         *
-         * @deprecated Types are in the process of being removed, use {@link Item(String, String)} instead.
-         */
-        @Deprecated
-        public Item(@Nullable String index, @Nullable String type, String id) {
-            if (id == null) {
-                throw new IllegalArgumentException("Item requires id to be non-null");
-            }
-            this.index = index;
-            this.type = type;
-            this.id = id;
-        }
-
-        /**
-         * Constructor for an artificial document request, that is not present in the index.
-         *
-         * @param index the index to be used for parsing the doc
-         * @param type the type to be used for parsing the doc
-         * @param doc the document specification
-         *
-         * @deprecated Types are in the process of being removed, use {@link Item(String, XContentBuilder)} instead.
-         */
-        @Deprecated
-        public Item(@Nullable String index, @Nullable String type, XContentBuilder doc) {
-            if (doc == null) {
-                throw new IllegalArgumentException("Item requires doc to be non-null");
-            }
-            this.index = index;
-            this.type = type;
-            this.doc = BytesReference.bytes(doc);
-            this.xContentType = doc.contentType();
-        }
-
         /**
          * Read from a stream.
          */
         Item(StreamInput in) throws IOException {
             index = in.readOptionalString();
-            type = in.readOptionalString();
+            if (in.getVersion().before(Version.V_2_0_0)) {
+                // types no longer supported so ignore
+                String type = in.readOptionalString();
+                if (type != null) {
```

Comment:
This is in the query DSL; for same reason as above I think we want the ISE since it's a breaking change and types have been deprecated for the entire 7.x/1.x line?

Reply:
:+1: leaving it up to you, not throwing ISE would (probably) bring less surprises at runtime during migration